### PR TITLE
Drop Node.js 14 and 16

### LINKED ISSUE #3746: Should node 14 really be supported?
### ⚠️ Please verify that this bug has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/uptime-kuma/security/policy)

### Description

Node.js 14, 16 reached end-of-life already. Should they really be supported? They use OpenSSL 1.1.1 which is EOL as well.

I think it's better to force node>=18 in `package.json` and drop related tests.

### 👟 Reproduction steps

Use node 14

### 👀 Expected behavior

Exit

### 😓 Actual Behavior

It works

### 🐻 Uptime-Kuma Version

master branch

### 💻 Operating System and Arch

All

### 🌐 Browser

All

### 🐋 Docker Version

_No response_

### 🟩 NodeJS Version

_No response_

### 📝 Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by harryzcy**
Tick the checkbox if you understand [x]: 
- [x] I have read and understand the pull request rules.

# Description

Fixes #3746

## Type of change

Please delete any options that are not relevant.

- Other

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas
  (including JSDoc for methods)
- [x] My changes generate no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

## Screenshots (if any)

Please do not use any external image service. Instead, just paste in or drag and drop the image here, and it will be uploaded automatically.

### CODE REVIEW

**Review by Saibamen — APPROVED** [review]

**Comment by harryzcy** [comment]
Is there anything blocking on this?

**Comment by Zaid-maker** [comment]
> Is there anything blocking on this?

Could please update Nodejs to `v18.8.2`?

**Comment by chakflying** [comment]
Even though they are officially out of support, there are still many devices out there using old versions. I think I don't see the need to drop testing for them unless there is a feature that we really need only in newer versions.

**Comment by CommanderStorm** [comment]
@harryzcy 
Thank you for this contribution.
Currently, I don't really see a reason to force our users to upgrade.
If there are (security/other) reasons which we are missing, we would love to hear them either in this issue (if not security related) or as a security advisory if it is.

Note that we will likely update to node >=16 as part of the gamedig-5 update as mentioned in https://github.com/louislam/uptime-kuma/issues/4414

Personal note:
We might have to change this to only support the versions of node in security support, depending what comes out of (^= we will see in a few years) the `EU Product Liability Directive`[^1] and depending on what "commercial" really means.

[^1]: See 
https://fosdem.org/2024/schedule/event/fosdem-2024-3683-the-regulators-are-coming-one-year-on/ for a good introduction to this topic

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -230,7 +230,7 @@ The goal is to make the Uptime Kuma installation as easy as installing a mobile
 
 ## Tools
 
-- [`Node.js`](https://nodejs.org/) >= 14
+- [`Node.js`](https://nodejs.org/) >= 18
 - [`npm`](https://www.npmjs.com/) >= 8.5
```
Comment (Saibamen) [inline CONTRIBUTING.md:234]:
```suggestion
- [`npm`](https://www.npmjs.com/) >= 10.2.4
```

File: README.md [hunk 2]
```diff
@@ -56,7 +56,7 @@ Requirements:
   - ✅ Major Linux distros such as Debian, Ubuntu, CentOS, Fedora and ArchLinux etc.
   - ✅ Windows 10 (x64), Windows Server 2012 R2 (x64) or higher
   - ❌ Replit / Heroku
-- [Node.js](https://nodejs.org/en/download/) 14 / 16 / 18 / 20.4
+- [Node.js](https://nodejs.org/en/download/) 18 / 20.4
 - [npm](https://docs.npmjs.com/cli/) 9
```
Comment (Saibamen) [inline README.md:60]:
```suggestion
- [npm](https://docs.npmjs.com/cli/) 10
```

**Review by Saibamen — CHANGES_REQUESTED** [review]
Latest NodeJS 18.x includes NPM 10.2.4

See: https://nodejs.org/en/download

File: CONTRIBUTING.md (hunk 1 above)
Comment (chakflying) [reply CONTRIBUTING.md:234]: We are currently using `lockfileVersion: 3`. I think as long as people are using an npm version that generates this version of lockfile, it should be fine. From the docs it became the default starting from v9.

https://docs.npmjs.com/cli/v10/configuring-npm/package-lock-json#handling-old-lockfiles

**Comment by CommanderStorm** [comment]
@Saibamen unsure how to proceeded:
image

=> revert bumping to `npm@9`?

**Comment by Saibamen** [comment]
Try to force download latest 18.x
image

Set this option to true:

`check-latest: true`

File: .github/workflows/auto-test.yml [hunk 3]
```diff
@@ -61,7 +61,7 @@ jobs:
         uses: actions/setup-node@v4
         with:
           node-version: ${{ matrix.node }}
```
Comment (Saibamen) [inline .github/workflows/auto-test.yml:63]:
```suggestion
          node-version: ${{ matrix.node }}
          check-latest: true
```

File: .github/workflows/auto-test.yml [hunk 4]
```diff
@@ -33,13 +33,10 @@ jobs:
...
       with:
         node-version: ${{ matrix.node }}
-  
-    - name: Install
-      run: npm install
-    - name: Build
-      run: npm run build
-    - name: Test Backend
-      run: npm run test-backend
+    - run: npm install npm@10 -g
```
Comment (CommanderStorm) [inline .github/workflows/auto-test.yml:36]:
```suggestion
    - run: npm install npm@9 -g
```

File: .github/workflows/auto-test.yml [hunk 5]
```diff
@@ -64,6 +61,8 @@ jobs:
         uses: actions/setup-node@v4
         with:
           node-version: ${{ matrix.node }}
+          check-latest: true
+      - run: npm install npm@10 -g
```
Comment (CommanderStorm) [inline .github/workflows/auto-test.yml:65]:
```suggestion
      - run: npm install npm@9 -g
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -230,8 +230,8 @@ The goal is to make the Uptime Kuma installation as easy as installing a mobile
 
 ## Tools
 
-- [`Node.js`](https://nodejs.org/) >= 14
-- [`npm`](https://www.npmjs.com/) >= 8.5
+- [`Node.js`](https://nodejs.org/) >= 18
+- [`npm`](https://www.npmjs.com/) >= 10.2.4
```
Comment (CommanderStorm) [inline CONTRIBUTING.md:234]:
```suggestion
- [`npm`](https://www.npmjs.com/) >= 9.3
```

File: README.md [hunk 7]
```diff
@@ -56,14 +56,14 @@ Requirements:
   - ✅ Major Linux distros such as Debian, Ubuntu, CentOS, Fedora and ArchLinux etc.
   - ✅ Windows 10 (x64), Windows Server 2012 R2 (x64) or higher
   - ❌ Replit / Heroku
-- [Node.js](https://nodejs.org/en/download/) 14 / 16 / 18 / 20.4
-- [npm](https://docs.npmjs.com/cli/) 9
+- [Node.js](https://nodejs.org/en/download/) 18 / 20.4
+- [npm](https://docs.npmjs.com/cli/) 10
```
Comment (CommanderStorm) [inline README.md:60]:
```suggestion
- [npm](https://docs.npmjs.com/cli/) 9
```

File: README.md [hunk 8]
```diff
@@ -56,14 +56,14 @@ Requirements:
...
 - [Git](https://git-scm.com/downloads)
 - [pm2](https://pm2.keymetrics.io/) - For running Uptime Kuma in the background
 
 ```bash
 # Update your npm
-npm install npm@9 -g
+npm install npm@10 -g
```
Comment (CommanderStorm) [inline README.md:66]:
```suggestion
npm install npm@9 -g
```

**Review by CommanderStorm — COMMENTED** [review]
reverting the changes to the npm version as our readme also does not specify a minimum node version => sticking to the minimum npm version for the given node version

**Comment by CommanderStorm** [comment]
I think we have discussed that we need to drop node 14+16 for v2 in #4557

@chakflying @louislam any concearns about merging this PR?

**Comment by louislam** [comment]
`npm@9` should be not necessary anymore.

**Comment by chakflying** [comment]
To summarize, moving to

>= v16 gives us:

- built-in parsing of certificates (#3942)
- gamedig `5.0`

>= v18 gives us:

- croner `8.0`

**Comment by Zaid-maker** [comment]
Louis is right npm@9 not necessary

**Comment by CommanderStorm** [comment]
just to recap: `npm@9` was added because of the lockfile version of v3, right?

**Comment by CommanderStorm** [comment]
What breaking changes? `ci-name`[^1] is not used and the other changes are not really relevant to us.

[^1]: https://docs.npmjs.com/cli/v10/using-npm/changelog#1000-2023-08-31

**Comment by louislam** [comment]
As I remembered, Node.js 14 or 16 on GitHub Actions came with npm 6 or 7 before which was too old. So I added the `@latest` tag in order to update the npm to the latest version (v9 at that moment). 

But later, the `@latest` became v10 which is not compatible with Node.js 14/16, so I locked it to `@9`.

**Pull request merged**